tests/utils/network_setup.go

```diff
@@ -0,0 +1,276 @@
+
+	log.Info(
+		"Created URIs for subnets",
+		"chainAURIs", chainANodeURIs,
+		"chainBURIs", chainBNodeURIs,
+		"blockchainIDA", blockchainIDA,
+		"blockchainIDB", blockchainIDB,
+	)
+
+	chainAWSURI := HttpToWebsocketURI(chainANodeURIs[0], blockchainIDA.String())
+	log.Info("Creating ethclient for blockchainA", "wsURI", chainAWSURI)
+	chainAWSClient, err = ethclient.Dial(chainAWSURI)
+	Expect(err).Should(BeNil())
+
+	chainAIDInt, err = chainAWSClient.ChainID(context.Background())
+	Expect(err).Should(BeNil())
+
+	chainBWSURI := HttpToWebsocketURI(chainBNodeURIs[0], blockchainIDB.String())
+	log.Info("Creating ethclient for blockchainB", "wsURI", chainBWSURI)
+	chainBWSClient, err = ethclient.Dial(chainBWSURI)
+	Expect(err).Should(BeNil())
+
+	chainBIDInt, err = chainBWSClient.ChainID(context.Background())
+	Expect(err).Should(BeNil())
+
+	log.Info("Finished setting up e2e test subnet variables")
+}
+
+// DeployTeleporterContract deploys the Teleporter contract to the two subnets. The caller is responsible for generating the
+// deployment transaction information
+func DeployTeleporterContract(transactionBytes []byte, deployerAddress common.Address, contractAddress common.Address) {
+	log.Info("Deploying Teleporter contract to subnets")
+
+	subnetsInfo := GetSubnetsInfo()
+
+	// Set the package level teleporterContractAddress
+	teleporterContractAddress = contractAddress
+
+	ctx := context.Background()
+
+	for _, subnetInfo := range subnetsInfo {
+		client := subnetInfo.ChainWSClient
+
+		chainID, err := client.ChainID(ctx)
+		Expect(err).Should(BeNil())
+		nonce, err := client.NonceAt(ctx, fundedAddress, nil)
+		Expect(err).Should(BeNil())
+		gasTipCap, err := client.SuggestGasTipCap(context.Background())
+		Expect(err).Should(BeNil())
+		baseFee, err := client.EstimateBaseFee(context.Background())
+		Expect(err).Should(BeNil())
+		gasFeeCap := baseFee.Mul(baseFee, big.NewInt(relayerEvm.BaseFeeFactor))
+		gasFeeCap.Add(gasFeeCap, big.NewInt(relayerEvm.MaxPriorityFeePerGas))
+		// Fund the deployer address
+		{
+			value := big.NewInt(0).Mul(big.NewInt(1e18), big.NewInt(10)) // 10eth
+			txA := types.NewTx(&types.DynamicFeeTx{
+				ChainID:   chainID,
+				Nonce:     nonce,
+				To:        &deployerAddress,
+				Gas:       DefaultTeleporterTransactionGas,
+				GasFeeCap: gasFeeCap,
+				GasTipCap: gasTipCap,
+				Value:     value,
+			})
+			txSigner := types.LatestSignerForChainID(chainID)
+			triggerTx, err := types.SignTx(txA, txSigner, fundedKey)
+
+			SendAndWaitForTransaction(ctx, subnetInfo.ChainWSClient, triggerTx)
+			receipt, err := client.TransactionReceipt(ctx, triggerTx.Hash())
+			Expect(err).Should(BeNil())
+			Expect(receipt.Status).Should(Equal(types.ReceiptStatusSuccessful))
+		}
+		log.Info("Finished funding Teleporter deployer")
+
+		// Deploy Teleporter
+		{
+			rpcClient, err := rpc.DialContext(ctx, HttpToRPCURI(subnetInfo.ChainNodeURIs[0], subnetInfo.BlockchainID.String()))
```

Comment:
I think we should include the RPC clients in `SubnetTestInfo`. There are a few useful rpc endpoints that we might want in tests.

Reply:
What are those use cases? For context, I removed the RPC clients from the tests altogether, since they are redundant with the websocket clients. Websocket clients are able to make RPC requests, but RPC clients aren't able to establish websocket connections. The `rpcClient` here is normally wrapped by `ethclient.Client`, but `ethclient.Client` doesn't have a way to send raw transaction bytes, so we have to do it this was here. Since for this test we're only doing that to issue the initial Teleporter deploy tx, there's no reason to expose the `rpcClient` to the tests themselves.

Reply:
`ethclient.Client` limits us to the enumerated function calls in that interface. Being able to call `rpcClient.CallContext` with `debug_traceTransaction` would have helped speed my debugging up quite a bit. I've been pausing the test and making those calls from the command line, but being able to do them programmatically would be helpful.

Reply:
I would argue that development/debugging tools shouldn't factor in strongly to an interface. When debugging, you can always construct the `rpcClient` from the `subnetInfo`. I would assume that debugging code would be removed before merging anyway.

Reply:
Yeah that's fair. The debugging code will obviously be removed from my test, but I'd like to leave behind a utility that can be used to debug easily in the future. I can do that without the RPC client being exposed so this is good 👍